pilot/pkg/model/network.go

```diff
@@ -545,7 +567,11 @@ func (n *networkGatewayNameCache) resolve(name string) ([]string, time.Duration)
 	wg.Wait()
 
 	sort.Strings(out)
-	return out, time.Duration(ttl) * time.Second
+	if errs.Len() == 2 {
+		// return error only if all requests are failed
+		err = errs
```

Comment:
nit: IMO using the named return vals is confusing, just make this `return out, ttl, nil`